### ISSUE TITLE
[PSOBB] Patch Server: Clients should be able to download patch files without a username/password

### ISSUE THREAD

**Issue by ShiftaDeband** [fuzziqersoftware/newserv#299]
Good evening,

If a user connects to newserv without a username and password saved and tries to 'Download Patch', they will be disconnected from the patch server. The output from the terminal indicates an issue with their serial number or username is missing, and disconnected them due to this fact. This appears to be a semi-recent change as this worked previously, though I do not have the commit that may have changed this.

If the user has a username and password, the patch server download proceeds as expected.

I believe that expected behavior should be that the user should be able to download patches without this information. 

A user must have a username and password saved to 'Start Game', though, which is on the client (and later, server to validate, which it should in the transition from patch server -> login server in the 'Start Game' scenario)

```
I 25498 2023-12-28 01:35:43 - [C-8] Created
I 25498 2023-12-28 01:35:43 - [Server] Client connected: C-8 on fd 38 via 26 (T-11000-PC_PATCH-bb-patch-patch_server_bb)
I 25498 2023-12-28 01:35:43 - [Commands] Sending to C-8 (version=BB_PATCH command=02 flag=00)
0000 | 4C 00 02 00 50 61 74 63 68 20 53 65 72 76 65 72 | L   Patch Server
0010 | 2E 20 43 6F 70 79 72 69 67 68 74 20 53 6F 6E 69 | . Copyright Soni
0020 | 63 54 65 61 6D 2C 20 4C 54 44 2E 20 32 30 30 31 | cTeam, LTD. 2001
0030 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0040 | 00 00 00 00 BE 2D 35 F8 B5 F7 37 76             |      -5   7v    
I 25498 2023-12-28 01:35:43 - [Commands] Received from C-8 (version=BB_PATCH command=02 flag=00)
0000 | 04 00 02 00                                     |                 
I 25498 2023-12-28 01:35:43 - [Commands] Sending to C-8 (version=BB_PATCH command=04 flag=00)
0000 | 04 00 04 00                                     |                 
I 25498 2023-12-28 01:35:43 - [Commands] Received from C-8 (version=BB_PATCH command=04 flag=00)
0000 | 70 00 04 00 00 00 00 00 00 00 00 00 00 00 00 00 | p               
0010 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0020 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0030 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0040 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0050 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0060 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
W 25498 2023-12-28 01:35:43 - [Server] Error processing client command: serial number is zero or username is missing
I 25498 2023-12-28 01:35:43 - [Server] Client disconnected: C-8 on fd 38
I 25498 2023-12-28 01:35:43 - [C-8] Deleted
```

**Comment by nolrinale** [comment]
This is incorrect behavior, you should be able to do Patch Download without need to input the login information ,and is a bug I noticed that also happens in Tethealla client.

I think the custom server IP addresses arent being set on first run for this option but only when you choose Start Game, then if you alt+backspace and select Patch Download it will connect to the custom patch server just fine.

**Comment by fuzziqersoftware** [comment]
This should be fixed by 09ac8921. I've tested the cases where a username and password is present, and where only a username is present, but I wasn't able to test the case where neither a username or password is present. Please try it again after this change and let me know if it works for you.

**Comment by ShiftaDeband** [comment]
This is fixed, thank you.

The way I've been able to reliably clear the username and password is to uncheck the box to save username/password in options.exe, though I've seen this not apply properly before.

The settings that I'm seeing are saved to the registry under `HKEY_CURRENT_USER\Software\SonicTeam\PSOBB` and the `ACCOUNT_CHECK` 32-bit DWORD set to 1 saves the username/password, setting it to 0 does not save username/password. (It looks like the US and JP client share these flags if you have both installed.)
